### ISSUE TITLE
Feature request: add `valid-test-locks` rule

### ISSUE THREAD

**Issue by cpAdm** [playwright-community/eslint-plugin-playwright#508]
## Feature request

Add a `playwright/valid-test-locks` rule, analogous to the existing `valid-test-tags` rule, to validate Playwright Test `lock` values against a project-defined allowlist or denylist.

### Motivation

Locks, introduced in Playwright v1.63, serialize tests that share an external resource. A typo silently creates a different lock name, defeating the intended serialization while leaving the test otherwise valid. A lint rule can catch this without requiring any Playwright runtime or TypeScript API changes.

This follows on from microsoft/playwright#42593, which requested compile-time lock-name constraints. That issue was closed without an implementation in Playwright itself.

### Proposed configuration

```js
{
  rules: {
    'playwright/valid-test-locks': ['error', {
      allowedLocks: ['account', 'billing'],
      // disallowedLocks: ['legacy-resource'],
    }],
  },
}
```

The rule should validate string literals and arrays of string literals in the details objects accepted by both tests and suites:

```ts
test('updates an account', { lock: 'account' }, async () => {});
test.describe('billing', { lock: ['billing'] }, () => {});

// Report: `payments` is not an allowed lock.
test('invalid lock', { lock: 'payments' }, async () => {});
```

Dynamic expressions could be ignored, consistent with the limits of static analysis. The option names and semantics should mirror `valid-test-tags` where practical.
